src/Compilers/CSharp/Portable/Symbols/TypeWithState.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Roslyn.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// A type and its corresponding flow state resulting from evaluating an rvalue expression.
+    /// </summary>
+    [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
+    internal readonly struct TypeWithState
+    {
+        public TypeSymbol Type { get; }
+        public NullableFlowState State { get; }
+        public bool HasNullType => Type is null;
+        public bool MaybeNull => State == NullableFlowState.MaybeNull;
```

Comment:
`IsMaybeNull`

Reply:
`MayBeNull`?

---
In reply to: [263853738](https://github.com/dotnet/roslyn/pull/33927#discussion_r263853738) [](ancestors = 263853738)

Reply:
Discussed in person `IsMaybeNull` doesn't make a lot of sense when you say it out loud. Keeping as is.